### ISSUE TITLE
start service on ios 13.2

### ISSUE THREAD

**Issue by yunmengzehaha** [libimobiledevice-win32/imobiledevice-net#135]
Does anyone know how to start service on ios 13 device. Any help will be greatly appreciated.

**Comment by qmfrederik** [comment]
Normally you'd do something like this:

```csharp
var api = LibiMobileDevice.Instance;

// Connect to your device. You'll need a connection to the lockdown daemon running on the device,
// which is your entrypoint to any other service to which you want to connect
int count = 0;
api.iDevice.idevice_get_device_list(out var udids, ref count);
api.iDevice.idevice_new(out var iDeviceHandle, udids[0])
lockdown.lockdownd_client_new_with_handshake(iDeviceHandle, out var lockdownClientHandle, "Label");

// Connect to your service. Use the name of the service you want to connect to.
const string serviceName = "com.apple.webinspector";

// Start the service on the device. You can also use 
// lockdownd_start_service_with_escrow_bag if needed
this.api.Lockdown.lockdownd_start_service(lockdownClientHandle, serviceName, out var service).ThrowOnError();

// You can now use api.Service.service_receive_with_timeout and api.Service.service_send
// to interact with the service on the device
```

imobiledevice-net is fairly up to date with upstream libimobiledevice. If you have a specific use case which does not work, feel free to share that here.

**Comment by yunmengzehaha** [comment]
Hi, thanks for your reply. I used a service "com.apple.instruments.remoteserver", I searched that this service uses SSL connection on iOS 13. When I use "com.apple.instruments.server.services.deviceinfo" channel to get some device info. I can get nothing. When I use iOS 12 device, it runs well. Some of the code are as follows:
`LockdownClientHandle lockdownHandle;
            var error = lockdown.lockdownd_client_new_with_handshake(deviceHandle, out lockdownHandle, "Quamotion");
            Console.WriteLine("Error_1: {0}", error);
            LockdownServiceDescriptorHandle service1;
            error = lockdown.lockdownd_start_service(lockdownHandle, "com.apple.instruments.remoteserver", out service1);
            Console.WriteLine("Error_2: {0}", error);
            var service_error = service.service_client_new(deviceHandle, service1, out client);
            Console.WriteLine("Service error: {0}", service_error);`
I'm not sure whether "service_enable_ssl" function can do any help.

**Comment by yunmengzehaha** [comment]
> Hi, thanks for your reply. I used a service "com.apple.instruments.remoteserver", I searched that this service uses SSL connection on iOS 13. When I use "com.apple.instruments.server.services.deviceinfo" channel to get some device info. I can get nothing. When I use iOS 12 device, it runs well. Some of the code are as follows:
> `LockdownClientHandle lockdownHandle;
> 
> var error = lockdown.lockdownd_client_new_with_handshake(deviceHandle, out lockdownHandle, @@"Quamotion");
> 
> Console.WriteLine("Error_1: {0}", error);
> 
> LockdownServiceDescriptorHandle service1;
> 
> error = lockdown.lockdownd_start_service(lockdownHandle, "com.apple.instruments.remoteserver", out service1);
> 
> Console.WriteLine("Error_2: {0}", error);
> 
> var service_error = service.service_client_new(deviceHandle, service1, out client);
> 
> Console.WriteLine("Service error: {0}", service_error);`
> 
> I'm not sure whether "service_enable_ssl" function can do any help. @qmfrederik

**Comment by qmfrederik** [comment]
@yunmengzehaha You are correct that `com.apple.instruments.remoteserver` requires SSL to be enabled.

`com.apple.instruments.remoteserver` uses a rather complex protocol, though.

**Comment by yunmengzehaha** [comment]
> @yunmengzehaha You are correct that `com.apple.instruments.remoteserver` requires SSL to be enabled.
> 
> `com.apple.instruments.remoteserver` uses a rather complex protocol, though.

The problem is that I find
